agr_java_core/src/main/java/org/alliancegenome/core/translators/document/GeneTranslator.java

```diff
@@ -156,30 +158,35 @@ protected GeneDocument entityToDocument(Gene entity, int translationDepth) {
                             orth.isBestRevScore(),
                             orth.getConfidence(),
                             orth.getGene1().getSpecies() == null ? null : orth.getGene1().getSpecies().getPrimaryKey(),
-                                    orth.getGene2().getSpecies() == null ? null : orth.getGene2().getSpecies().getPrimaryKey(),
-                                            orth.getGene1().getSpecies() == null ? null : orth.getGene1().getSpecies().getName(),
-                                                    orth.getGene2().getSpecies() == null ? null : orth.getGene2().getSpecies().getName(),
-                                                            orth.getGene1().getSymbol(),
-                                                            orth.getGene2().getSymbol(),
-                                                            orth.getGene1().getPrimaryKey(),
-                                                            orth.getGene2().getPrimaryKey(),
-                                                            notCalled, matched, notMatched
-                            );
+                            orth.getGene2().getSpecies() == null ? null : orth.getGene2().getSpecies().getPrimaryKey(),
+                            orth.getGene1().getSpecies() == null ? null : orth.getGene1().getSpecies().getName(),
+                            orth.getGene2().getSpecies() == null ? null : orth.getGene2().getSpecies().getName(),
+                            orth.getGene1().getSymbol(),
+                            orth.getGene2().getSymbol(),
+                            orth.getGene1().getPrimaryKey(),
+                            orth.getGene2().getPrimaryKey(),
+                            notCalled, matched, notMatched
+                    );
                     olist.add(doc);
                 }
 
             }
             geneDocument.setOrthology(olist);
         }
 
-        if (entity.getDiseaseEntityJoins() != null && translationDepth > 0) {
-            List<DiseaseDocument> diseaseList = diseaseTranslator.getDiseaseDocuments(entity, entity.getDiseaseEntityJoins(), translationDepth);
+        if (gene.getDiseaseEntityJoins() != null && translationDepth > 0) {
+            List<DiseaseDocument> diseaseList = diseaseTranslator.getDiseaseDocuments(gene, gene.getDiseaseEntityJoins(), translationDepth);
             geneDocument.setDiseases(diseaseList);
         }
 
-        if (entity.getGenomeLocations() != null) {
+        if (gene.getPhenotypeEntityJoins() != null && gene.getPhenotypeEntityJoins().size() > 0 && translationDepth > 0) {
+            List<PhenotypeDocument> phenotypeList = phenotypeTranslator.getPhenotypeDocuments(gene, gene.getPhenotypeEntityJoins(), translationDepth);
```

Comment:
This caused the indexer to die, and I would guess this was untested code considering the data was not there to back it up.

Reply:
translationDepth should have been (translationDepth - 1)